addons/postgresql/scripts/etcd-clean.sh

```diff
@@ -1,34 +1,50 @@
 #!/bin/bash
-
 if [ -z "$ETCD_SERVER" ]; then
   exit 0
 fi
 
-IFS=',' read -ra ADDR <<< "$KB_CLUSTER_COMPONENT_POD_IP_LIST"
-FIRST_IP=${ADDR[0]}
+echo "Find command kubectl"
+if command -v kubectl &> /dev/null; then
+  IFS=',' read -ra POD_NAMES <<< "$KB_CLUSTER_POD_NAME_LIST"
 
-PATRONI_API_URL="http://$FIRST_IP:8008/patroni"
-PATRONI_SHUTDOWN_URL="http://$FIRST_IP:8008/shutdown"
+  for POD_NAME in "${POD_NAMES[@]}"; do
+    echo "Attempting to execute pkill -f /usr/local/bin/patroni in pod: $POD_NAME"
 
-status_code=$(curl -s -o /dev/null -w "%{http_code}" $PATRONI_API_URL)
+    MAX_ATTEMPTS=3
+    ATTEMPT=1
+    SUCCESS=0
+    
+    while [ "$ATTEMPT" -le "$MAX_ATTEMPTS" ]; do
+      echo "Attempt $ATTEMPT: Executing pkill -f /usr/local/bin/patroni in pod: $POD_NAME"
+      kubectl exec $POD_NAME -- pkill -f /usr/local/bin/patroni
```

Comment:
A particularly challenging corner case arises when a Patroni instance becomes unresponsive due to pod failure, process timeout, or network partition (which is even more difficult to handle), causing its shutdown API to fail. In such situations, how should the etcd-clean process proceed? 

Under normal circumstances, etcd-clean should report an error and exit, waiting for the next retry of the preTerminate task. 

This introduces two key requirements:

- etcd-clean needs to be idempotent. This means that Patroni processes that were shut down in the previous round of etcd-clean should be detected and skipped in subsequent rounds.
- The stop cluster/component API should include a force flag. This flag would allow the operation to bypass the preTerminate step and release resources in cases where preTerminate execution fails. In the context of etcd-clean, this failure might occur due to Patroni processes not being completely terminated, potentially leaving uncleaned garbage objects in the shared etcd.